pkg/kv/kvserver/rangefeed/catchup_scan.go

```diff
@@ -119,6 +119,8 @@ func (i *CatchUpIterator) Close() {
 // TODO(ssd): Clarify memory ownership. Currently, the memory backing
 // the RangeFeedEvents isn't modified by the caller after this
 // returns. However, we may revist this in #69596.
+// TODO(dt): Does this really need to be a pointer to a struct containing all
+// pointers? can we pass by value instead?
```

Comment:
I think so? Eventually this gets copied into a kvpb.MuxRangeFeedEvent which stores it by-value.

Reply:
Maybe what I really want is BulkEvents -> BulkValues -- and just emit rangekeys as we hit them -- if that would mean the signature could be []kv.KeyValue to match the Result.Rows from a ScanRequest like the one we use for init-scan and is expected by OnValues().

Anyway, this can be a follow-up: we can reallocate this slice ten times to change what is or isn't a pointer and still come out ahead over individual deliveries.

Reply:
Yeah, apologies, my only point was that I think this was a pointer simply because of function signatures and that we have some by-value uses in the call stack so I don't think much thought was really put into this.  I agree your potential follow up might be good.